pipe/section/section_impls/postgres_connector/src/destination.rs

```diff
@@ -52,27 +58,37 @@ impl Postgres {
                     };
                     let name = escape_table_name(message.origin());
                     let mut transaction = connection.begin().await?;
+                    let mut initialized = false;
+                    let mut insert_query = String::new();
                     while let Some(chunk) = message.next().await? {
                         let df = match chunk{
                             Chunk::DataFrame(df) => df,
                             _ => Err("expected dataframe chunk".to_string())?
                         };
-                        let schema = generate_schema(name.as_str(), df.as_ref())?;
                         let mut columns = df.columns();
-                        sqlx::query(&schema).execute(&mut *transaction).await?;
-                        let values_placeholder = columns.iter().enumerate()
-                            .map(|(pos, col)|  {
-                                let suffix = match col.data_type() {
-                                    DataType::RawJson => "::json",
-                                    _ => "",
-                                };
-                                format!("${}{}", pos + 1, suffix)
-                            })
-                            .collect::<Vec<_>>()
-                            .join(",");
-                        let insert = format!("INSERT INTO \"{name}\" VALUES({values_placeholder})");
+                        if !initialized {
+                            initialized = true;
+                            let schema = generate_schema(self.schema.as_str(), name.as_str(), df.as_ref())?;
+                            sqlx::query(&schema).execute(&mut *transaction).await?;
+                            if self.truncate {
+                                sqlx::query(&format!("TRUNCATE \"{}\".\"{name}\"", self.schema))
+                                    .execute(&mut *transaction)
+                                    .await?;
+                            }
+                            let values_placeholder = columns.iter().enumerate()
```

Comment:
It might be helpful to leave a short comment about why this is changing datatypes.

Reply:
marking as resolved